orbit/pkg/update/runner.go

```diff
@@ -144,7 +146,21 @@ func (r *Runner) HasLocalHash(target string) bool {
 func (r *Runner) Execute() error {
 	log.Debug().Msg("start updater")
 
-	ticker := time.NewTicker(r.opt.CheckInterval)
+	// Randomize the initial interval so that all agents don't synchronize their updates
+	initialInterval := r.opt.CheckInterval
+	// Assuming CheckInterval > 0 because we validated it in NewRunner
+	nBig, err := rand.Int(rand.Reader, big.NewInt(int64(r.opt.CheckInterval)))
+	if err != nil {
+		log.Info().Err(err).Msg("randomization of initial update interval failed")
+	} else {
+		initialInterval = time.Duration(nBig.Int64())
```

Comment:
Wouldn't it be better to ADD the random value to the initial time rather than reduce it?

Reply:
I thought of that, but that seemed dishonest. Someone might complain that checking for the update took longer than specified.